src/api/supportRequest.ts

```diff
@@ -12,6 +12,34 @@ export interface NextSupportRequestResponse {
   userNotified: boolean;
 }
 
+supportRequestRoutes.get('/getAll', async (req, res) => {
+
+  let status = req.query.status;
+  var requests;
+  if(status)
+  {
+    switch(status){
+      case "Abandoned":
+        requests = await SupportRequest.find({ status: SupportRequestStatus.Abandoned });
+        res.send(requests);
+      case "Complete":
+        requests = await SupportRequest.find({ status: SupportRequestStatus.Complete });
+        res.send(requests);
+      case "Pending":
+        requests = await SupportRequest.find({ status: SupportRequestStatus.Pending });
+        res.send(requests);
+      case "InProgress":
+        requests = await SupportRequest.find({ status: SupportRequestStatus.InProgress });
+        res.send(requests);
+      default:
+        res.send("It doesn't look like that's a valid status!");
```

Comment:
It's super important with situations like this that you follow HTTP standards to communicate with the front end, so we need to update this to send a status back to the front end. In this situation, the frontend goofed up in making the request, so the status should indicate that the client was at fault. I'll let you figure out what status to use. I think the correct syntax is `res.status(XXX).send('Invalid status')`